### DISCUSSION TITLE
Should I create multiple physical tables or let partition handle it?

### DISCUSSION THREAD

**Discussion by rajnish-aargo** [ClickHouse/ClickHouse#34170] [Q&A] [answered]
Hello,
I am thinking of using clickhouse for our time series data. But I am not familiar with deep internals of architecture yet. 
It's a humongous data set (in Peta Bytes). Let's take below representative logical table schema for my data
Vendor | City | Time | DataField1 | DataField2 | ....
There are couple hundred vendors and each vendor contains few thousand cities (with some minor overlap between vendors which I guess is not relevant here)
My question is, should I create a single table with above schema or create multiple tables, say 1 per Vendor or even 1 per vendor per City? What's the trade off here? 
1) I can see multiple tables might save some storage as then then vendor/city columns with lot of duplicate entries go away, but ClickHouse compression might take care of that in case of single table.
2) I think bigger difference might be in query time as I will be using Vendor and City to Index the data. Then question would be is finding a table more costly op than finding right chunk of data in a single table (need to have PARTITION, ORDER property in schema)?
3) Which one would scale better? Which is easier to manage in sharding/cluster setting?
Thanks in advance!

**Comment by den-crane** [comment, marked as answer]
single table `partition by toYYYYMM(event_time) order by (tenant_id,..)`

https://github.com/ClickHouse/ClickHouse/issues/31919

https://stackoverflow.com/a/70714188/11644308

**Comment by rajnish-aargo** [comment]
Thank you!
And seems like I shouldn't partition by Vendor and City either, I should just ORDER BY (Vendor, City).
And is it a thumb rule to always partition by TIME? And is there some good number of rows for a partition? I see this lot of this toYYYYMM() but I guess it should be about number of rows, a month might have just 1000 entries or even 1 Billion.

**Comment by den-crane** [comment]
Initially (before Dec. 2017) CH has only one possible partition key - month.

A thumb rule , the insert should insert into only one partition. Number of inserts is up to 1 insert **per second PER INSTANCE.** (NOT PER TABLE).
During day ingestion should not insert into "OLD / ALL" partitions. 

Personally I use `PARTITION BY (account_id,toYYYYMM (access_day))` **BUT** I wrote sophisticated ETL and I have maximum 40 possible accounts.
